types/react-native-push-notification/index.d.ts

```diff
@@ -1,34 +1,36 @@
-// Type definitions for react-native-push-notification 5.0
+// Type definitions for react-native-push-notification 7.0
 // Project: https://github.com/zo0r/react-native-push-notification#readme
 // Definitions by: Paito Anderson <https://github.com/PaitoAnderson>
 //                 Tom Sawkins <https://github.com/tomSawkins>
 //                 Andrew Li <https://github.com/Li357>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 3.5
 
 export interface PushNotificationPermissions {
     alert?: boolean;
     badge?: boolean;
     sound?: boolean;
 }
 
-export interface PushNotification {
+export interface ReceivedNotification {
     foreground: boolean;
     userInteraction: boolean;
     message: string | object;
-    data: object;
+    data: Record<string, any>;
+    userInfo: Record<string, any>;
     subText?: string;
     badge: number;
     alert: object;
     sound: string;
     id: number;
+    action?: string;
     finish: (fetchResult: string) => void;
 }
 
 export interface PushNotificationOptions {
     onRegister?: (token: { os: string; token: string }) => void;
-    onNotification?: (notification: PushNotification) => void;
-    onAction?: (notification: PushNotification) => void;
+    onNotification?: (notification: Omit<ReceivedNotification, 'userInfo'>) => void;
+    onAction?: (notification: Omit<ReceivedNotification, 'data'>) => void;
```

Comment:
Not completely sure if `data` should be omitted here. See [this line](https://github.com/zo0r/react-native-push-notification/blob/3641bf66cc6147715a272069167cf48de3d2a00e/index.js#L335). It doesn't seem like the `data` property is deleted. But `notificationData` is defined nowhere, so maybe it always fails silently with the try/catch (bug in the package)?  In any case, the `data` property still seems to be in the notification.

Reply:
It seems it's a bug in the package. I'll revert omitting `data`.